### PR TITLE
fix(cssnano-preset): disable z-index minification

### LINKED ISSUE #7548: CSS minification changes z-index
### Have you read the Contributing Guidelines on issues?

- [X] I have read the [Contributing Guidelines on issues](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#reporting-new-issues).

### Description

Hey Docusaurus Team,

thanks for your amazing work :)

    

We realized recently that docusaurus cleans up z-indices in our custom css that we are loading via `customCss: require.resolve('./src/css/custom.css')`. My question is: **How can I disable the z-index modification?** This might be a good feature but can lead to bugs when you are using other third party that bring their own css. We are using `'@docusaurus/preset-classic'`.

### Self-service

- [ ] I'd be willing to address this documentation request myself.

### PR DESCRIPTION
**Pull request by dpang314**
## Pre-flight checklist

- [x] I have read the [Contributing Guidelines on pull requests](https://github.com/facebook/docusaurus/blob/main/CONTRIBUTING.md#pull-requests).
- [ ] **If this is a code change**: I have written unit tests and/or added dogfooding pages to fully verify the new behavior.
- [ ] **If this is a new API or substantial change**: the PR has an accompanying issue (closes #7548) and the maintainers have approved on my working plan.

## Motivation

`z-index` minification can cause issues when working with the styles of third party libraries. 

## Test Plan

On the [dogfooding page](https://deploy-preview-7593--docusaurus-2.netlify.app/tests/pages/z-index-tests/), the text would normally have its ` z-index` minified to 3, but after these changes, it remains 100.

For comparison, one place on the Docusaurus website that already uses `z-index` is on the [showcase page](https://docusaurus.io/showcase)
1. Go to https://docusaurus.io/showcase and open Chrome DevTools with responsive dimensions and device set to Desktop
2. Hover over a tag (e.g. "Favorite" or "Open-Source") and then right click on it
3. If you navigate to the element in DevTools (it should have the role "tooltip")
4. On the [official site](https://docusaurus.io/showcase) the z-index will be 7, but on the [site deploy of this pull request](https://deploy-preview-7593--docusaurus-2.netlify.app/showcase), the z-index is 500, which is what it is defined as in the CSS file.

### Test links

Dogfooding: https://deploy-preview-7593--docusaurus-2.netlify.app/tests/pages/z-index-tests/
Deploy preview: https://deploy-preview-7593--docusaurus-2.netlify.app/

## Related issues/PRs

#7548

### CODE REVIEW

File: website/docs/cli.md [hunk 1]
```diff
@@ -92,7 +92,9 @@ Compiles your site for production.
...
+<!-- cSpell:ignore zindex -->
+
+For advanced minification of CSS bundle, we use the [advanced cssnano preset](https://github.com/cssnano/cssnano/tree/master/packages/cssnano-preset-advanced) with [postcss-zindex](https://github.com/cssnano/cssnano/tree/master/packages/postcss-zindex) disabled. We also use several additional PostCSS plugins and [level 2 optimization of clean-css](https://github.com/jakubpawlowicz/clean-css#level-2-optimizations). If as a result of this advanced CSS minification you find broken CSS, build your website with the environment variable `USE_SIMPLE_CSS_MINIFIER=true` to minify CSS with the [default cssnano preset](https://github.com/cssnano/cssnano/tree/master/packages/cssnano-preset-default). **Please [fill out an issue](https://github.com/facebook/docusaurus/issues/new?labels=bug%2C+needs+triage&template=bug.md) if you experience CSS minification bugs.**
```
Comment (Josh-Cena) [inline website/docs/cli.md:97]: No need to document this; we just change it however we want to

**Review by Josh-Cena — APPROVED** [review]
LGTM; will wait for @slorber's opinion on this. I don't think sane applications would have much bloat caused by z-index anyway (can't speak for those who add a 0 every time something doesn't work).

**Comment by slorber** [comment]
👍 thanks, looks like useful to disable this

It's marked as unsafe in CSSNano doc: https://cssnano.co/docs/optimisations/zindex/

Considering it's hard to ensure users will use be in the safe use-case, let's turn it off.

**Comment by slorber** [comment]
👍 thanks, looks like useful to disable this

It's marked as unsafe in CSSNano doc: https://cssnano.co/docs/optimisations/zindex/

Considering it's hard to ensure users will use be in the safe use-case, let's turn it off.

**Pull request merged**